src/coreclr/jit/patchpoint.cpp

```diff
@@ -151,8 +151,21 @@ class PatchpointTransformer
 
         helperBlock->SetFlags(BBF_BACKWARD_JUMP | BBF_NONE_QUIRK);
 
-        compiler->fgAddRefPred(helperBlock, block);
-        compiler->fgAddRefPred(remainderBlock, helperBlock);
+        FlowEdge* const falseEdge = compiler->fgAddRefPred(helperBlock, block);
+        FlowEdge* const trueEdge  = compiler->fgGetPredForBlock(remainderBlock, block);
+
+        if (trueEdge->hasLikelihood())
+        {
+            falseEdge->setLikelihood(1.0 - trueEdge->getLikelihood());
+        }
+        else
+        {
+            trueEdge->setLikelihood(0.5);
+            falseEdge->setLikelihood(0.5);
```

Comment:
The test actually passing should be very rare (I believe the threshold is 1/10000). Also, isn't `trueEdge` always going to be the `BBJ_ALWAYS` edge inserted by `fgSplitBlockAtBeginning` above, so the check here is a bit meaningless?

Reply:
> Also, isn't trueEdge always going to be the BBJ_ALWAYS edge inserted by fgSplitBlockAtBeginning above, so the check here is a bit meaningless?

Ah you're right, so `trueEdge` should always have an edge likelihood here. I'll remove the branch.

> The test actually passing should be very rare (I believe the threshold is 1/10000).

Is there any value to being more precise than 0.1/0.9 for `trueEdge`/`falseEdge`'s likelihoods?

Reply:
In tier0? Doubtful. Also 1/10000 is not entirely accurate since I believe multiple patchpoints share the counter, but @AndyAyersMS can correct me if I'm wrong...

Reply:
I see below this snippet that `helperBlock` inherits 1% of `block`'s weight; should we have the edge likelihoods match that? So `trueEdge` is 0.99, and `falseEdge` is 0.01.

Reply:
Makes sense to me to be consistent with that. On further look it looks like `TC_OnStackReplacement_InitialCounter` is 1000. Not sure if the mismatch is intentional or not; in a way, the most accurate guess we could make is probably `1 / <that value>`.

Reply:
Overall it probably doesn't matter since (at least for now) we only have patchpoints in unoptimized code. I guess I'd go with the precedent already there and use the 0.99/0.01 split.